packages/vscode-messenger-common/src/messages.ts

```diff
@@ -176,8 +176,113 @@ export type NotificationHandler<P> = (params: P, sender: MessageParticipant) =>
  * Base API for Messenger implementations.
  */
 export interface MessengerAPI {
-    sendRequest<P, R>(type: RequestType<P, R>, receiver: MessageParticipant, params?: P): Promise<R>
+    sendRequest<P, R>(type: RequestType<P, R>, receiver: MessageParticipant, params?: P, cancelable?: CancellationToken): Promise<R>
     onRequest<P, R>(type: RequestType<P, R>, handler: RequestHandler<P, R>): void
     sendNotification<P>(type: NotificationType<P>, receiver: MessageParticipant, params?: P): void
     onNotification<P>(type: NotificationType<P>, handler: NotificationHandler<P>): void
 }
+
```

Comment:
I propose to move the following definitions to a separate file `cancellation.ts`. This one is about messages, so we can keep the CancelRequest related definitions here.

Reply:
Moved